### ISSUE TITLE
Cube Store doesn't support certain pre-aggregations

### ISSUE THREAD

**Issue by denis-stafichuk-parsable** [cube-js/cube.js#3100]
**Describe the bug**
Cube Store doesn't support certain pre-aggregations:
```
Error while querying: scheduler-a0987e44-080a-4551-b842-d46422bfd2c1 (40ms)
{
  "processingId": 46,
  "queueSize": 1,
  "queryKey": [
    [
      "CREATE TABLE dev_pre_aggregations.job_funnel_job_cohorts AS SELECT\n      \"job_funnel\".step \"job_funnel__step\", \"jobs\".team_id \"jobs__team_id\", date_trunc('day', (\"job_funnel\".t::timestamptz AT TIME ZONE 'UTC')) \"job_funnel__time_day\", count(\"job_funnel\".user_id) \"job_funnel__conversions\"\n    FROM\n      (WITH joined_events AS (\n    select\n    scheduled_events.user_id scheduled_user_id,\ndone_events.user_id done_user_id,\nscheduled_events.t\n    FROM\n(select id user_id, (to_timestamp(created_at) AT TIME ZONE 'UTC') t from (select id, created_at from public.jobs where job_type = 0) e) scheduled_events\nLEFT JOIN\n(select id user_id, (to_timestamp(created_at) AT TIME ZONE 'UTC') t from (select id, created_at from public.jobs where job_type = 0 and completed_at > 0) e) done_events ON scheduled_events.user_id = done_events.user_id AND done_events.t >= scheduled_events.t\n  )\n  select user_id, first_step_user_id, step, max(t) t from (\n    SELECT scheduled_user_id user_id, scheduled_user_id first_step_user_id, t, 'Scheduled' step FROM joined_events\nUNION ALL\nSELECT done_user_id user_id, scheduled_user_id first_step_user_id, t, 'Done' step FROM joined_events\n  ) as event_steps GROUP BY 1, 2, 3) AS \"job_funnel\"\nLEFT JOIN public.jobs AS \"jobs\" ON \"job_funnel\".user_id = \"jobs\".id  GROUP BY 1, 2, 3",
      []
    ],
    [
      [
        {
          "refresh_key": "162611851"
        }
      ],
      [
        {
          "refresh_key": "162611851"
        }
      ]
    ]
  ],
  "queuePrefix": "SQL_PRE_AGGREGATIONS_STANDALONE_default",
  "timeInQueue": 0
} 
Error: Error during create table: CREATE TABLE dev_pre_aggregations.job_funnel_job_cohorts_vrekpfsb_wjm4qfl4_1gep6bg (`job_funnel__step` varchar(255), `jobs__team_id` uuid, `job_funnel__time_day` timestamp, `job_funnel__conversions` bigint)  LOCATION ?: User: Create table failed: Internal: Invalid byte 45, offset 8.
    at WebSocket.<anonymous> (/cube/node_modules/@cubejs-backend/cubestore-driver/src/WebSocketConnection.ts:87:30)
    at WebSocket.emit (events.js:314:20)
    at Receiver.receiverOnMessage (/cube/node_modules/ws/lib/websocket.js:873:20)
    at Receiver.emit (events.js:314:20)
    at Receiver.dataMessage (/cube/node_modules/ws/lib/receiver.js:502:14)
    at Receiver.getData (/cube/node_modules/ws/lib/receiver.js:435:17)
    at Receiver.startLoop (/cube/node_modules/ws/lib/receiver.js:143:22)
    at Receiver._write (/cube/node_modules/ws/lib/receiver.js:78:10)
    at doWrite (_stream_writable.js:403:12)
    at writeOrBuffer (_stream_writable.js:387:5)
    at Receiver.Writable.write (_stream_writable.js:318:11)
    at Socket.socketOnData (/cube/node_modules/ws/lib/websocket.js:948:35)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:297:12)
    at readableAddChunk (_stream_readable.js:272:9)
    at Socket.Readable.push (_stream_readable.js:213:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
```

**To Reproduce**
1. Clone https://github.com/denis-stafichuk-parsable/cube-store-bug repo I created to demonstrate the issue
2. Run `docker compose up -d` to spin up cube.js backend and PostgreSQL
3. Run `curl 'http://localhost:4000/cubejs-api/v1/load?query=%7B%22measures%22%3A%5B%22JobFunnel.conversions%22%5D%2C%22timeDimensions%22%3A%5B%7B%22dimension%22%3A%22JobFunnel.time%22%2C%22dateRange%22%3A%5B%222021-06-12%22%2C%222021-07-12%22%5D%7D%5D%2C%22order%22%3A%7B%22JobFunnel.conversions%22%3A%22desc%22%7D%2C%22dimensions%22%3A%5B%22JobFunnel.step%22%5D%2C%22filters%22%3A%5B%5D%7D&queryType=multi'` to load the data
4. Get
```
{
  "error": "Error: Error during create table: CREATE TABLE dev_pre_aggregations.job_funnel_job_cohorts_gcoqhyh3_wjm4qfl4_1gepb4g (`job_funnel__step` varchar(255), `jobs__team_id` uuid, `job_funnel__time_day` timestamp, `job_funnel__conversions` bigint)  LOCATION ?: User: Create table failed: Internal: Invalid byte 45, offset 8."
}
```
as a response and see the error above in logs
5. `git checkout postgres-pre-aggregations`
6. Run `curl 'http://localhost:4000/cubejs-api/v1/load?query=%7B%22measures%22%3A%5B%22JobFunnel.conversions%22%5D%2C%22timeDimensions%22%3A%5B%7B%22dimension%22%3A%22JobFunnel.time%22%2C%22dateRange%22%3A%5B%222021-06-12%22%2C%222021-07-12%22%5D%7D%5D%2C%22order%22%3A%7B%22JobFunnel.conversions%22%3A%22desc%22%7D%2C%22dimensions%22%3A%5B%22JobFunnel.step%22%5D%2C%22filters%22%3A%5B%5D%7D&queryType=multi'` to load the data
7. Get correct response without any errors in logs:
```
{
  "queryType": "regularQuery",
  "results": [
    {
      "query": {
        "measures": [
          "JobFunnel.conversions"
        ],
        "timeDimensions": [
          {
            "dimension": "JobFunnel.time",
            "dateRange": [
              "2021-06-12T00:00:00.000",
              "2021-07-12T23:59:59.999"
            ]
          }
        ],
        "order": [
          {
            "id": "JobFunnel.conversions",
            "desc": true
          }
        ],
        "dimensions": [
          "JobFunnel.step"
        ],
        "filters": [],
        "timezone": "UTC",
        "queryType": "regularQuery"
      },
      "data": [
        {
          "JobFunnel.step": "Scheduled",
          "JobFunnel.conversions": "4"
        },
        {
          "JobFunnel.step": "Done",
          "JobFunnel.conversions": "1"
        }
      ],
      "lastRefreshTime": "2021-07-12T20:47:13.497Z",
      "refreshKeyValues": [],
      "usedPreAggregations": {
        "dev_pre_aggregations.job_funnel_job_cohorts": {
          "targetTableName": "dev_pre_aggregations.job_funnel_job_cohorts_o0qs1iww_wjm4qfl4_1gepai6"
        }
      },
      "transformedQuery": {
        "sortedDimensions": [
          "JobFunnel.step"
        ],
        "sortedTimeDimensions": [
          [
            "JobFunnel.time",
            "day"
          ]
        ],
        "measures": [
          "JobFunnel.conversions"
        ],
        "leafMeasureAdditive": true,
        "leafMeasures": [
          "JobFunnel.conversions"
        ],
        "hasNoTimeDimensionsWithoutGranularity": false,
        "allFiltersWithinSelectedDimensions": true,
        "isAdditive": true,
        "granularityHierarchies": {
          "year": [
            "year",
            "month",
            "day",
            "hour",
            "minute",
            "second"
          ],
          "week": [
            "week",
            "day",
            "hour",
            "minute",
            "second"
          ],
          "month": [
            "month",
            "day",
            "hour",
            "minute",
            "second"
          ],
          "day": [
            "day",
            "hour",
            "minute",
            "second"
          ],
          "hour": [
            "hour",
            "minute",
            "second"
          ],
          "minute": [
            "minute",
            "second"
          ]
        },
        "hasMultipliedMeasures": false
      },
      "annotation": {
        "measures": {
          "JobFunnel.conversions": {
            "title": "Job Funnel Conversions",
            "shortTitle": "Conversions",
            "type": "number",
            "drillMembers": [],
            "drillMembersGrouped": {
              "measures": [],
              "dimensions": []
            }
          }
        },
        "dimensions": {
          "JobFunnel.step": {
            "title": "Job Funnel Step",
            "shortTitle": "Step",
            "type": "string"
          }
        },
        "segments": {},
        "timeDimensions": {}
      },
      "dataSource": "default",
      "dbType": "postgres",
      "extDbType": "cubestore",
      "external": false,
      "slowQuery": false
    }
  ],
  "pivotQuery": {
    "measures": [
      "JobFunnel.conversions"
    ],
    "timeDimensions": [
      {
        "dimension": "JobFunnel.time",
        "dateRange": [
          "2021-06-12T00:00:00.000",
          "2021-07-12T23:59:59.999"
        ]
      }
    ],
    "order": [
      {
        "id": "JobFunnel.conversions",
        "desc": true
      }
    ],
    "dimensions": [
      "JobFunnel.step"
    ],
    "filters": [],
    "timezone": "UTC",
    "queryType": "regularQuery"
  },
  "slowQuery": false
}
```
8. The only difference between the branches is where pre-aggregation is stored https://github.com/denis-stafichuk-parsable/cube-store-bug/compare/cube-store-pre-aggregations...postgres-pre-aggregations

**Expected behavior**
Get correct result without any errors in logs.

**Version:**
cubejs/cube:v0.27.50

**Comment by ovr** [comment]
Hello @denis-stafichuk-parsable ,

Fixed & released in `v0.27.52`. Can you verify that it works for you?

Thanks

**Comment by denis-stafichuk-parsable** [comment]
Hey @ovr,

It's working now!

Thanks a lot for such a quick fix!
